pinot-query-runtime/src/main/java/org/apache/pinot/query/service/dispatch/QueryDispatcher.java

```diff
@@ -85,15 +87,15 @@ public QueryDispatcher() {
   }
 
   public ResultTable submitAndReduce(long requestId, DispatchableSubPlan dispatchableSubPlan,
-      MailboxService mailboxService, long timeoutMs,
+      MailboxService mailboxService, OpChainSchedulerService scheduler, long timeoutMs,
       Map<String, String> queryOptions, Map<Integer, ExecutionStatsAggregator> executionStatsAggregator,
       boolean traceEnabled)
       throws Exception {
     try {
       // submit all the distributed stages.
       int reduceStageId = submit(requestId, dispatchableSubPlan, timeoutMs, queryOptions);
       // run reduce stage and return result.
-      return runReducer(requestId, dispatchableSubPlan, reduceStageId, timeoutMs, mailboxService,
+      return runReducer(requestId, dispatchableSubPlan, reduceStageId, timeoutMs, mailboxService, scheduler,
```

Comment:
@ankitsultana @tibrewalpratik17 can you guys verify if this resolves the broker busy wait issue you saw from your end?

Reply:
I think it should fix the issue that we were seeing internally.

Just a side-note: it would be interesting to see how this scales for Pinot v1 engine use-cases with high-qps. The current design there is a simple request/response model which would likely be faster than repeatedly yielding and processing results. We can visit this later though.